grr/server/grr_response_server/gui/ui/components/flow_args_form/osquery_form.ts

```diff
@@ -26,24 +26,23 @@ export class OsqueryForm extends FlowArgumentForm<OsqueryArgs> implements
     query: new FormControl(this.defaultQueryDisplayed, Validators.required),
     timeoutMillis: new FormControl(null, Validators.required),
     ignoreStderrErrors: new FormControl(null),
-    fileCollectColumns: new FormControl([]),
+    fileCollectionColumns: new FormControl([]),
   });
 
   fileCollectionSettingsShown = false;
   lowLevelSettingsShown = false;
 
   readonly separatorKeysCodes: number[] = [ENTER, COMMA];
 
-  get fileCollectColumnsControl() {
-    const control = this.form.get('fileCollectColumns');
-    if (!control) {
-      throw Error('fileCollectColumns form control is null');
-    }
+  get fileCollectionColumnsControl(): AbstractControl {
+    const control = this.form.get('fileCollectionColumns');
+    assertNonNull(control,
+        'the form control listing columns for file collection');
 
     return control;
   }
-  get fileCollectColumns() {
-    return this.fileCollectColumnsControl?.value as Array<string>;
+  get fileCollectionColumns(): Array<string> {
```

Comment:
For an array, please use `string[]`. Also, can we narrow it down to ReadonlyArray?

Reply:
We can't narrow it to `ReadonlyArray` in the current implementation - I am mutating this array: https://github.com/google/grr/blob/d64fafe58ebb3a964b9dcf6d6315bd646aef8cba/grr/server/grr_response_server/gui/ui/components/flow_args_form/osquery_form.ts#L87

I looked online but didn't find a guideline for dealing with arrays within form controls (or other mutable data types). An alternative to the current approach would be to make a copy of the array every time and use `form.patchValue` (i.e. using "Readonly" arrays).

Also I couldn't find enough documentation for what actions I should take when mutating the value of a control in this way --- if I should call [markAsDirty](https://angular.io/api/forms/AbstractControl#markasdirty) (which I am doing in the current implementation), or [updateValueAndValidity](https://angular.io/api/forms/AbstractControl#updatevalueandvalidity) or both...

Reply:
Thanks for the clarification! In another component we just use push without calling `markAsDirty` or anything else afterwards (see https://github.com/google/grr/blob/252af5d85f02121a75f8d83ebbee8e3d7e4f5fc9/grr/server/grr_response_server/gui/ui/components/flow_args_form/collect_multiple_files_form.ts#L54). The code seems to work but I wonder if that's a coincidence or an intended behavior.

Using `patchValue` seems to be the bullet-proof solution in this case, to be honest, but I don't have a strong opinion on this.

Reply:
Thinking about it now, immutable arrays and `patchValue` seem like the best option - we wouldn't rely on shady Angular features, and it wouldn't hurt performance as we only expect few columns (at most 5, and in most cases 0 or 1). I will refactor it and then you can take a final look.

Reply:
Sgtm!